WooCommerce/Classes/Blaze/BlazeCampaignCreationCoordinator.swift

```diff
@@ -183,6 +184,31 @@ private extension BlazeCampaignCreationCoordinator {
         blazeNavigationController.viewControllers = [controller]
         navigationController.present(blazeNavigationController, animated: true, completion: nil)
     }
+
+    func presentNoProductAlert() {
+        let alert = UIAlertController(title: Localization.NoProductAlert.title,
+                                      message: Localization.NoProductAlert.message,
+                                      preferredStyle: .alert)
+        let createAction = UIAlertAction(title: Localization.NoProductAlert.createProduct, style: .default) { [weak self] _ in
+            self?.startProductCreation()
+        }
+        let cancelAction = UIAlertAction(title: Localization.NoProductAlert.cancel, style: .cancel)
+        alert.addAction(createAction)
+        alert.addAction(cancelAction)
+        navigationController.dismiss(animated: true) { [weak self] in
+            self?.navigationController.present(alert, animated: true)
+        }
+    }
+
+    func startProductCreation() {
+        let coordinator = AddProductCoordinator(siteID: siteID,
+                                                source: .blazeCampaignCreation,
+                                                sourceView: nil,
+                                                sourceNavigationController: navigationController,
+                                                isFirstProduct: true)
+        self.addProductCoordinator = coordinator
```

Comment:
nit: `self` not needed here.

Reply:
Removed in 67dee68d7f26976f5ef40b53e1769c560eccc9d8